### ISSUE TITLE
Huge memory consumption on cursor leader node

### ISSUE THREAD

**Issue by aversant** [liftbridge-io/liftbridge#332]
After starting the process, the memory consumption stops at 1-1,5 Gb. But on the cursor leader node the consumption continues to grow and reaches 10 GB on `liftbridge` proccess.
When the cursor leader changes, the growth stops, but resumes on the new leader node. In this case, the memory on the former leader is not released.

`User Memory graph`. Green line on right side - change cursor leader.

![image](https://user-images.githubusercontent.com/18608678/111023577-b1ec1080-83fb-11eb-88d7-03950257f6f8.png)

Result profiling `mevent01`: (`go tool pprof -alloc_objects /bin/liftbridge http://127.0.0.1:8080/debug/pprof/heap`):

```go
(pprof) top
Showing nodes accounting for 11244192778, 53.49% of 21020572605 total
Dropped 447 nodes (cum <= 105102863)
Showing top 10 nodes out of 165
      flat  flat%   sum%        cum   cum%
3029887270 14.41% 14.41% 6072457649 28.89%  github.com/liftbridge-io/liftbridge/server/commitlog.(*index).ReadEntryAtFileOffset
1530281031  7.28% 21.69% 1530281031  7.28%  bytes.NewReader (inline)
1512289348  7.19% 28.89% 1512289348  7.19%  encoding/binary.Read
1150716201  5.47% 34.36% 1150716201  5.47%  google.golang.org/grpc/internal/transport.(*itemList).enqueue (inline)
1050860348  5.00% 39.36% 3253041140 15.48%  google.golang.org/grpc/internal/transport.(*http2Server).operateHeaders
 810575917  3.86% 43.22%  810575917  3.86%  google.golang.org/grpc/internal/transport.(*decodeState).addMetadata (inline)
 717531840  3.41% 46.63%  717531840  3.41%  google.golang.org/protobuf/internal/impl.(*MessageInfo).MessageOf
 578021747  2.75% 49.38%  578021747  2.75%  golang.org/x/net/http2.(*Framer).readMetaFrame.func1
 457330327  2.18% 51.56% 1472396521  7.00%  github.com/liftbridge-io/liftbridge/server.(*apiServer).Publish
 406698749  1.93% 53.49% 1144435264  5.44%  golang.org/x/net/http2.(*Framer).readMetaFrame
```

mevent02:

```go
(pprof) top 50
Showing nodes accounting for 111827139838, 93.29% of 119872726229 total
Dropped 480 nodes (cum <= 599363631)
Showing top 50 nodes out of 117
      flat  flat%   sum%        cum   cum%
30542608604 25.48% 25.48% 61102386487 50.97%  github.com/liftbridge-io/liftbridge/server/commitlog.(*index).ReadEntryAtFileOffset
15281379094 12.75% 38.23% 15281379094 12.75%  bytes.NewReader (inline)
15278398789 12.75% 50.97% 15278398789 12.75%  encoding/binary.Read
3933181144  3.28% 54.25% 3933181144  3.28%  google.golang.org/grpc/internal/transport.(*itemList).enqueue (inline)
2958729281  2.47% 56.72% 2958729281  2.47%  github.com/liftbridge-io/liftbridge/server/commitlog.SerializedMessage.Headers
2923137708  2.44% 59.16% 2923137708  2.44%  github.com/liftbridge-io/liftbridge/server.eventTimestampsToProto
2875698018  2.40% 61.56% 2875698018  2.40%  google.golang.org/protobuf/internal/impl.(*MessageInfo).MessageOf
2392065240  2.00% 63.56% 4451894130  3.71%  google.golang.org/grpc/internal/transport.(*http2Server).Write
1956872729  1.63% 65.19% 6096093985  5.09%  google.golang.org/grpc/internal/transport.(*http2Server).operateHeaders
1859455535  1.55% 66.74% 4317891542  3.60%  github.com/liftbridge-io/liftbridge/server.natsToProtoMessage
1666594549  1.39% 68.13% 5992519686  5.00%  github.com/liftbridge-io/liftbridge/server/commitlog.newMessageSetFromProto
1657667730  1.38% 69.51% 3503442321  2.92%  encoding/binary.Write
1525340583  1.27% 70.78% 1525340583  1.27%  google.golang.org/grpc/internal/transport.(*decodeState).addMetadata (inline)
1476679603  1.23% 72.02% 1476679603  1.23%  github.com/liftbridge-io/liftbridge-api/go.(*Message).Unmarshal
1352870432  1.13% 73.14% 67452156128 56.27%  github.com/liftbridge-io/liftbridge/server.(*apiServer).subscribe.func1
1194058201     1% 74.14% 1194058201     1%  github.com/liftbridge-io/liftbridge-api/go.(*Message).Marshal

...............

```

`Profiling files`:

[mevent01_pprof.liftbridge.alloc_objects.alloc_space.inuse_objects.inuse_space.001.pb.gz](https://github.com/liftbridge-io/liftbridge/files/6134392/mevent01_pprof.liftbridge.alloc_objects.alloc_space.inuse_objects.inuse_space.001.pb.gz)
[mevent02_pprof.liftbridge.alloc_objects.alloc_space.inuse_objects.inuse_space.001.pb.gz](https://github.com/liftbridge-io/liftbridge/files/6134393/mevent02_pprof.liftbridge.alloc_objects.alloc_space.inuse_objects.inuse_space.001.pb.gz)

if you need more information - I can clarify. Thanks.

**Comment by tylertreat** [comment]
I haven't been able to reproduce this yet. I'm seeing the previous leader's memory drop back down.

**Comment by aversant** [comment]
Fixed on #351 .
